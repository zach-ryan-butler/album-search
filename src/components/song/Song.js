import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Song({ obj }) {
  return (
    <section>
      <Link to={`/lyrics/${obj.name}/${obj.album}/${obj.title}/${obj.id}`}>{obj.title}</Link>
    </section>
  );
}

Song.propTypes = {
  obj: PropTypes.shape({
    title: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    album: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
  }).isRequired,
};

export default Song;
