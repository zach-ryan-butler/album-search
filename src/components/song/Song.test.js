import React from 'react';
import { shallow } from 'enzyme';
import Song from './Song';

describe('Song component', () => {
  it('renders Song', () => {
    const wrapper = shallow(<Song
      obj={{
        title: 'some title',
        name: 'some name',
        album: 'some album',
        id: 'some id'
      }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
