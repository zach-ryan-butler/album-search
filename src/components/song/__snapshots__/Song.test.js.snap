// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song component renders Song 1`] = `
<section>
  <Link
    to="/lyrics/some name/some album/some title/some id"
  >
    some title
  </Link>
</section>
`;
